manifests/params.pp

```diff
@@ -106,6 +106,7 @@
         'php5-ldap',
         'php5-mysql',
         'php5-pgsql',
+        'mysql-client-core-5.5',
```

Comment:
yes I see the advantage of your point for cross platform.  From my side, I don't like / see the advantage of using a  module to install a simple client that is available already in a package. Then it's one less module to care about. 
That is why I mentioned the param way of installing or not a package. This  will make things simple I think.